src/cpp/session/modules/rmarkdown/SessionRMarkdown.cpp

```diff
@@ -1357,43 +1357,61 @@ void sourceCppConsoleOutputHandler(module_context::ConsoleOutputType type,
       LOG_ERROR(error);
 }
 
+Error prepareCacheConsoleOutputFile(const std::string& docId,
+                                    const std::string& chunkId,
+                                    const std::string& nbCtxId,
+                                    FilePath* pChunkOutputFile)
+{
+   // forward declare error
+   Error error;
+   
+   // prepare chunk directory
+   FilePath cachePath = notebook::chunkOutputPath(
+            docId,
+            chunkId,
+            notebook::ContextExact);
+   
+   error = cachePath.removeIfExists();
+   if (error)
+      return error;
+   
+   error = cachePath.ensureDirectory();
+   if (error)
+      return error;
+   
+   // prepare cache console output file
+   *pChunkOutputFile =
+         notebook::chunkOutputFile(docId, chunkId, nbCtxId, kChunkOutputText);
+   
+   return Success();
+}
+
 Error executeRcppEngineChunk(const std::string& docId,
                              const std::string& chunkId,
                              const std::string& nbCtxId,
                              const std::string& code,
+                             const std::map<std::string, std::string>& options,
                              json::JsonRpcResponse* pResponse)
 {
    // forward declare error
-   Error error = Success();
+   Error error;
    
    // always ensure we emit a 'execution complete' event on exit
    ChunkExecCompletedScope execScope(docId, chunkId);
    
-   // prepare chunk directory
-   FilePath outputPath = notebook::chunkOutputPath(
-            docId,
-            chunkId,
-            notebook::ContextExact);
-
-   error = outputPath.removeIfExists();
+   // prepare cache output file (use tempfile on failure)
+   FilePath targetPath = module_context::tempFile("rcpp-cache", "");
```

Comment:
@jjallaire if we fail to prepare a cache output file, we instead write to a temporary file and use that for (temporary) outputs. This should ensure the user still sees some appropriate error output even if something goes wrong internally.